src/ensembl/src/content/app/custom-download/components/content-builder/ContentBuilder.tsx

```diff
@@ -0,0 +1,263 @@
+import React from 'react';
+import CheckboxWithSelects from 'src/content/app/custom-download/components/checkbox-with-selects/CheckboxWithSelects';
+import CheckboxWithRadios from 'src/content/app/custom-download/components/checkbox-with-radios/CheckboxWithRadios';
+import CheckboxWithTextfields from 'src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields';
+
+import CheckboxGrid, {
+  CheckboxGridOption
+} from 'src/content/app/custom-download/components/checkbox-grid/CheckboxGrid';
+import { RadioOptions } from 'src/shared/radio/Radio';
+import { Option } from 'src/shared/select/Select';
+import {
+  Accordion,
+  AccordionItem,
+  AccordionItemHeading,
+  AccordionItemPanel,
+  AccordionItemButton
+} from 'src/shared/accordion';
+
+import styles from './ContentBuilder.scss';
+
+import get from 'lodash/get';
+import {
+  Attribute,
+  AttributeWithContent,
+  AttributeWithOptions
+} from 'src/content/app/custom-download/types/Attributes';
+import JSONValue, { PrimitiveOrArrayValue } from 'src/shared/types/JSON';
+
+type Path = (string | number)[];
+let path: Path = [];
+
+type ContentBuilderProps = {
+  data: AttributeWithContent;
+  selectedData: JSONValue;
+  onChange: (type: string, path: Path, payload: PrimitiveOrArrayValue) => void;
+  contentState: JSONValue;
+  onContentStateChange: (
+    type: string,
```

Comment:
The `type` argument doesn’t seem to be used by the callbacks in the parent components?

Reply:
It is not used now but I think it might be useful in the future to identify what type of data is getting stored.